docs/reference/contributing/target/sai.md

```diff
@@ -0,0 +1,100 @@
+- `sai_free()` deinitializes and unclocks the unused part of the device.
+- You can reinitialize a device or block by using `sai_init()` after you use `sai_free()` on it.
+
+If the device is a *receiver*:
+
+- `sai_transfer()` returns false if the `sai_t` object is NULL.
+- `sai_transfer()` returns false if there's no sample in the FiFo.
+- `sai_transfer()` if `psample` is NULL; it pops 1 sample from the FiFo and returns true.
+- `sai_transfer()` if `psample` is not NULL; it pops 1 sample from the FiFo, stores it to the address pointed by `psample` and returns true.
+
+If the device is a *transmitter*:
+
+- `sai_transfer()` returns false if the `sai_t` object is NULL.
+- `sai_transfer()` returns false if the FiFo is full and you can't push `*psample`.
+- `sai_transfer()` if `psample` is NULL; it pushes one '0' sample to the FiFo and returns true.
+- `sai_transfer()` if `psample` is not NULL; it pushes the pointed sample to the FiFo and returns true.
+
+#### Undefined behavior
+
+- Calling any function other than `sai_init()` before the initialization of the SAI.
+- Calling any function other than `sai_init()` after calling `sai_free()`.
+
+#### Other requirements
+
+A target must also define these elements to allow tests to be run:
+
+- `#define SAI_DEFAULT_SAMPLE_RATE (xxxxxU)`.
+- Pins for 2 SAI or I2S interfaces, including MCLK, BCLK, WCLK and SD named:
+  - SAI_A_MCLK, SAI_A_BCLK, SAI_A_WCLK and SAI_A_SD.
+  - SAI_B_MCLK, SAI_B_BCLK, SAI_B_WCLK and SAI_B_SD.
+
+#### Notes
+
+Watch out for this common trouble area when implementing this API. You should consider a transceiver supporting asynchronous rx/tx as 2 different peripherals:
+
+   - One read-only.
+   - One write-only.
+
+The first allocated channel may or may not limit the second one's feature. For example, in a peripheral that supports asynchronous rx/tx but requires the format of both instances to be the same, the first allocated instance sets the format and ties the second one to this format.
+
+### Dependencies
+
+Hardware SAI/I2S capabilities.
+
+### Implementing the SAI API
+
+You can find the API and specification for the SAI API in the following header file:
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://os-doc-builder.test.mbed.com/docs/development/feature-hal-spec-sai-doxy/classmbed_1_1_s_a_i.html)
+
+To enable SAI support in Mbed OS, add the `SAI` label in the `device_has` option of the target's section in the `targets.json` file.
+
+### Testing
+
+The Mbed OS HAL provides a set of conformance tests for SAI. You can use these tests to validate the correctness of your implementation. To run the SAI HAL tests, use the following command:
+
+```
+mbed test -t <toolchain> -m <target> -n "tests-mbed_hal-sai*"
+```
+
+You can read more about the test cases:
+
+[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/feature-hal-spec-sai/group__hal__sai__tests.html)
```

Comment:
Note: This link 404s, and I can't find the page in the Doxy.

Reply:
No because I'm generating the feature-branch Doxy here: https://os-doc-builder.test.mbed.com/docs/development/feature-hal-spec-sai-doxy/annotated.html

But I still don't see it. Will it be under "Data Structures" like the API is? Or is it somewhere else like "Files"?

Reply:
do you have this http://os-doc-builder.test.mbed.com/docs/development/feature-hal-spec-sai/group__hal_sai_tests.html ?

Reply:
When I click on that, it 404s

Reply:
`DEVICE_SAI` was missing from `mbed-os/doxyfile_options` that's probably why these sections were not generated.